src/Query/QueryBuilder.php

```diff
@@ -1920,7 +1921,7 @@ public function buildWithQueries($withs, &$params): string
     }
 
     /**
-     * @return ConnectionInterface|null the database connection.
+     * @psalm-return Connection
```

Comment:
That looks suspicious.

Reply:
Not because the interface does not have all the methods, you can indicate the implementation, it is a valid psalm annotation, but it will give an error that the method does not exist in the interface.